File: tests/smoke/spring-mybatis-ntier/src/main/java/com/example/order/controller/OrderController.java
package com.example.order.controller;

import com.example.order.service.OrderService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class OrderController {
    private final OrderService orderService;

    public OrderController(OrderService orderService) {
        this.orderService = orderService;
    }

    @GetMapping("/api/orders/{id}")
    public Object getOrder(@PathVariable Long id) {
        return orderService.findById(id);
    }

    @PostMapping("/api/orders")
    public Object createOrder(@RequestBody Object dto) {
        return orderService.create(dto);
    }
}

File: tests/smoke/nestjs-prisma/src/order/order.controller.ts
import { Controller, Get, Param, Post, Body } from '@nestjs/common';
import { OrderService } from './order.service';

@Controller('api/orders')
export class OrderController {
  constructor(private readonly orderService: OrderService) {}

  @Get(':id')
  findOne(@Param('id') id: string) {
    return this.orderService.findById(Number(id));
  }

  @Post()
  create(@Body() dto: Record<string, unknown>) {
    return this.orderService.create(dto);
  }
}
